src/dsl/cb.rs

```diff
@@ -66,14 +94,24 @@ pub fn and<F: From<u64>, E: Into<Constraint<F>>, I: IntoIterator<Item = E>>(
 
     for constraint in inputs.into_iter() {
         let constraint = constraint.into();
-        annotations.push(constraint.annotation);
-
-        expr = expr * constraint.expr;
+        match constraint.typing {
+            Typing::AntiBooly => {
+                annotations.push(constraint.annotation);
+                let constraint_expr = if constraint.expr == 0u64.expr() {1u64.expr()} else {0u64.expr()};
```

Comment:
The constraint expression doesn't have to be a constant, it could include a query. Hence, when AntiBooly we can only produce an error.

Reply:
Fixed. I mistaken !AntiBooly as the inverse of AntiBooly, and hence tried to create a Boolean inverse of it.